### PR TITLE
env: add win32 to supported platform environments (CRAFT-498)

### PR DESCRIPTION
**Pull request by cjp256**
Add a platform check for Linux before vetting that Charmcraft is
running from the snap (or running in developer mode).

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

**Review by facundobatista — CHANGES_REQUESTED** [review]
Please add a test for this, thanks!

**Comment by cjp256** [comment]
I reworked this and updated tests.

**Review by cmatsuoka — APPROVED** [review]
Looks good, with Facundo's suggested change.

File: charmcraft/env.py [hunk 1]
```diff
@@ -59,7 +59,12 @@ def is_charmcraft_running_in_managed_mode():
 
 def is_charmcraft_running_in_supported_environment():
     """Check if Charmcraft is running in a supported environment."""
-    return sys.platform == "linux" and is_charmcraft_running_from_snap()
+    if sys.platform == "linux":
+        return is_charmcraft_running_from_snap()
+    elif sys.platform in ["win32"]:
```
[the remark this answers is not in the document]
Comment (cjp256) [reply charmcraft/env.py:64]: I expect darwin to be added to that list. I just didn't want to do it without further testing :D

[the remark this answers is not in the document]
Comment (cjp256) [reply charmcraft/env.py]: done.

**Review by facundobatista — APPROVED** [review]
Thanks!

**Pull request merged**